### PR TITLE
Replacement tuning, auto-plunge on startup, manual testing with mcd-cli...

### PR DESCRIPTION
**Pull request by EdNoepel**
...also:
- allow bidding reservoir to empty completely (useful for `flap` auctions)
- removed unmaintained testchain scripts

### CODE REVIEW

File: auction_keeper/gas.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def get_gas_price(self, time_elapsed: int) -> Optional[int]:
             initial_price = int(round(fast_price * self.initial_multiplier))
 
         return GeometricGasPrice(initial_price=initial_price,
-                                 every_secs=30,
+                                 every_secs=42,
```
Comment (EdNoepel) [inline auction_keeper/gas.py:76]: Again, we don't want this configurable to constrain config complexity and prevent inappropriate values.

**Review by ith-harvey — APPROVED** [review]
I really like the plunge feature... Great as always Ed

File: auction_keeper/main.py [hunk 2]
```diff
@@ -318,6 +320,16 @@ def approve(self):
...
             self.collateral.approve(self.our_address, gas_price=self.gas_price)
 
+    def plunge(self):
+        pending_txes = get_pending_transactions(self.web3)
+        if len(pending_txes) > 0:
+            while len(pending_txes) > 0:
+                logging.warning(f"Cancelling first of {len(pending_txes)} pending transactions")
+                pending_txes[0].cancel(gas_price=self.gas_price)
+                # After the synchronous cancel, wait to see if subsequent transactions get mined
+                time.sleep(28)
```
Comment (KentonPrescott) [inline auction_keeper/main.py:330]: Is 28 seconds enough time? 

Fast price with `ethgasstation` is expected to be [mined within two minutes](https://docs.ethgasstation.info/gas-price), so IIUC, if the `gas-initial-multiplier` is 1, there's a possibility this doesn't wait long enough for a pending transaction to be overridden.

**Review by KentonPrescott — COMMENTED** [review]
Nice work on the auto-pluger logic.

Out of curiosity, why was support for a local testchain taken out and replaced with kovan testing?

**Comment by EdNoepel** [comment]
> Nice work on the auto-pluger logic.
> 
> Out of curiosity, why was support for a local testchain taken out and replaced with kovan testing?

Mostly because it was neither being used nor maintained.  Digging deeper, it had limited utility because parity doesn't simulate the amount of time a tx sits pending in the mempool as a function of gas price.

File: auction_keeper/main.py (hunk 2 above)
Comment (EdNoepel) [reply auction_keeper/main.py:330]: If the TX queue is stuck, later transactions would likely have already been replaced up to a large max gas limit (the default is 2000 GWEI).  Waiting two blocks should give these high-gas TXes a chance to be mined before pushing a cancellation TX with the same nonce onto the mempool.

File: auction_keeper/gas.py (hunk 1 above)
Comment (EdNoepel) [reply auction_keeper/gas.py:76]: Was actually on the fence with 56.  A smaller initial adjustment seemed prudent.

File: auction_keeper/main.py (hunk 2 above)
Comment (EdNoepel) [reply auction_keeper/main.py:330]: And if later transactions _didn't_ have high gas, we'd rather cancel than wait (seemingly) infinite time to free the TX queue.

**Pull request merged**